bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ItemRegistry.java

```diff
@@ -25,6 +25,14 @@
 public interface ItemRegistry extends Registry<Item, String> {
 
     /**
+     * This methods checks if an item is part of the registry.
+     *
+     * @param name the item name
+     * @return true if there is an item with given name, otherwise false
+     */
+    public boolean containsItem(String name);
```

Comment:
ItemRegistry extends Registry and thus has `public E get(K key);`, where K is String (the item name) and which returns null, if the element does not exist - you do not have to deal with ItemNotFoundExcpetion here.